### PR TITLE
Added code for workshop 4

### PR DESCRIPTION
**Pull request by MadsHeilmann**

### CODE REVIEW

**Comment by MadsHeilmann** [comment]
Hvordan får jeg clang tidy til at tie stille så jeg kan merge? :P

**Review by TexWO — DISMISSED** [review]

File: src/workshops/4/pgr.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+enum { TWO = 2, FIVE = 5, NINE = 9 };
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann https://google.github.io/styleguide/cppguide.html#Enumerator_Names

File: src/workshops/4/pgr.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/* Denne funktion skal returnere 1 hvis x er et primtal og 0 ellers */
+int isPrime(int number) {
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann https://google.github.io/styleguide/cppguide.html#Function_Names

Comment (audio-engineer) [reply src/workshops/4/pgr.c]: @MadsHeilmann You can return the `bool` data type in C23. https://en.cppreference.com/w/c/language/bool_constant
https://en.cppreference.com/w/c/language/arithmetic_types#Boolean_type

File: src/workshops/4/pgr.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if (number % i == 0) {
+      result = 0;
+      break;
+    }
+  }
+  return result;
+}
+
+/* Denne funktion skal returnere 1 hvis gcd(x,2)=1 og 0 ellers */
+int isGcd1(int number) { return number; }
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann `bool`

File: src/workshops/4/pgr.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+/* Denne funktion skal returnere 1 hvis 9^x-2 mod 5 = 2 og 0 ellers */
+int is2mod5(int number) {
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann `bool`

File: src/workshops/4/pgr.c
Comment (audio-engineer) [inline src/workshops/4/pgr.c:1]: @MadsHeilmann Let's call the file something more useful than `pgr.c`

File: src/workshops/4/pgr.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+enum { TWO = 2, FIVE = 5, NINE = 9 };
+
+/* Denne funktion skal returnere 1 hvis x er et primtal og 0 ellers */
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann https://www.doxygen.nl/manual/docblocks.html#cppblock

File: src/workshops/4/pgr.c [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  }
+  return result;
+}
+
+/* Denne funktion skal returnere 1 hvis gcd(x,2)=1 og 0 ellers */
+int isGcd1(int number) { return number; }
+
+/* Denne funktion skal returnere 1 hvis 9^x-2 mod 5 = 2 og 0 ellers */
+int is2mod5(int number) {
+  int result = ((int)pow(NINE, number % FIVE) - TWO) % FIVE == TWO ? 1 : 0;
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann Very difficult to read and understand. Also, if you want to have a real one-liner, no need to declare a variable first and then return it in line 25. You can just write:
```suggestion
return ((int)pow(NINE, number % FIVE) - TWO) % FIVE == TWO ? 1 : 0;
```
But please make it a bit more readable first.

File: src/workshops/4/pgr.c [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  return result;
+}
+
+int main(void) {
+  int number_to_test = 0;
+  int isPrime_bool = 0;
+  int isGcd1_bool = 0;
+  int is2mod5_bool = 0;
```
Comment (audio-engineer) [inline src/workshops/4/pgr.c]: @MadsHeilmann Very strange variable names which don't explain the intention very well. Try something like
```suggestion
  bool is_prime_number = false;
  bool is_greatest_common_divisor = false;
```

**Review by audio-engineer — CHANGES_REQUESTED** [review]

File: src/workshops/4/pgr.c (hunk 5 above)
Comment (MadsHeilmann) [reply src/workshops/4/pgr.c]: There are a few different styles in the link. Which do you think we should use?

File: src/workshops/4/pgr.c (hunk 7 above)
Comment (MadsHeilmann) [reply src/workshops/4/pgr.c]: Naming is updated. ClangTidy wants to initilize to "0" and not "false" so i have kept is as "0".

File: src/workshops/4/pgr.c (hunk 5 above)
Comment (audio-engineer) [reply src/workshops/4/pgr.c]: First one

File: src/workshops/4/conditions.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+/* Denne funktion skal returnere 1 hvis 9^x-2 mod 5 = 2 og 0 ellers */
+bool Is2Mod5(int number) {
+  int exponent_calculation = (int)pow(kNine, number % kFive);
+  // NOLINTNEXTLINE(readability-implicit-bool-conversion)
+  return (exponent_calculation - kTwo) % kFive == kTwo;
+}
+
+int main(void) {
+  int number_to_test = 0;
+  int is_prime_number = 0;
```
Comment (audio-engineer) [inline src/workshops/4/conditions.c]: Why not use the `bool` type here as well?

**Review by audio-engineer — CHANGES_REQUESTED** [review]

File: src/workshops/4/conditions.c (hunk 8 above)
Comment (MadsHeilmann) [reply src/workshops/4/conditions.c]: Do you know what the format specifier is for bool type? I can't seem to find it when googling.

Comment (audio-engineer) [reply src/workshops/4/conditions.c]: Idk, you can try to use `%d` or a ternary operator: https://www.programiz.com/c-programming/ternary-operator

**Review by audio-engineer — APPROVED** [review]

**Review by Danar435 — APPROVED** [review]

**Pull request merged**